### PR TITLE
Implement OAuthSecretConverter to support gloo mesh secret format

### PR DESCRIPTION
**Pull request by MitchAman**
# Description

This PR modifies the type of secret produced by `glooctl create secret oauth ...` from `Opaque` to `extauth.solo.io/oauth`. Opaque secrets that have been generated by prior releases may still be used successfully.

# Context

A new `SecretConverter` is added to the `GlooSecretConverterChain` which exclusively handles converting between k8s and gloo secret representations. Previously there was no converter that handled OAuth secrets, so the default `ResourceClient` handling was used: https://github.com/solo-io/solo-kit/blob/ddf6f96e8a26856642e094fa5bc8c7df0b4406f0/pkg/api/v1/clients/kubesecret/resource_client.go#L76-L97

The `OAuthSecretConverter` will only act on secrets of type `extauth.solo.io/oauth`. If the `OAuthSecretConverter` (and no other converters) do not act on the secret, then the default handling we were previously relying on will.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/api/converters/kube/oauth_secret_test.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		coreCache, err := cache.NewKubeCoreCache(ctx, clientset)
+		Expect(err).NotTo(HaveOccurred())
+		resourceClient, err = kubesecret.NewResourceClient(clientset, &v1.Secret{}, false, coreCache)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Describe("converting from a Kubernetes secret to Gloo", func() {
+
+		It("ignores secrets that aren't oauth", func() {
+			secret := &corev1.Secret{
```
Comment (bewebi) [inline projects/gloo/pkg/api/converters/kube/oauth_secret_test.go:55]: nit: could factor out `secret` declaration to `Describe` scope and only set the necessary fields in each `It`

File: projects/gloo/pkg/api/converters/kube/oauth_secret_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+			actual, err := converter.ToKubeSecret(ctx, resourceClient, &v1.Secret{
+				Metadata: &core.Metadata{Name: "foo"},
+				Kind:     &v1.Secret_Aws{},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual).To(BeNil())
+		})
+
+		It("correctly converts OAuth secrets", func() {
+			expected := &corev1.Secret{
```
Comment (bewebi) [inline projects/gloo/pkg/api/converters/kube/oauth_secret_test.go:121]: nit: `expected` is identical in this test and the next so we could/should factor out the declaration to the `Describe` scope

File: projects/gloo/pkg/api/converters/kube/secretconverter.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ var GlooSecretConverterChain = NewSecretConverterChain(
 	new(AwsSecretConverter),
 	new(HeaderSecretConverter),
 	new(APIKeySecretConverter),
+	new(OAuthSecretConverter),
```
Comment (bewebi) [inline projects/gloo/pkg/api/converters/kube/secretconverter.go:26]: nit: not sure it's in-scope for this PR, but I'm noticing that TLS and Aws converters live in this file while Header, APIKey, and now OAuth get their own files
ideally we'd factor out TLS and Aws to their own files as well for consistency

**Review by bewebi — COMMENTED** [review]
Looks good! Just some nits to take or leave

File: projects/gloo/pkg/api/converters/kube/secretconverter.go (hunk 3 above)
Comment (MitchAman) [reply projects/gloo/pkg/api/converters/kube/secretconverter.go:26]: I agree, we should be consistent, and haven't been.

I've got another small PR I'll be sending out related to the header secret types, so I'll address this comment in that PR

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**